pre-commit/yarn-lock-check.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+# Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set -euo pipefail
+
+TMP_DIR=$(mktemp -d)
+cp package.json yarn.lock $TMP_DIR
+
+(cd $TMP_DIR; yarn install --silent > /dev/null)
```

Comment:
if this script were only used from `pre-commit`, it could just be `yarn install`, since `pre-commit` fails upon any file changes. However, since I'm also calling it from `./fmt.sh --test`, which isn't supposed to modify any files (and I couldn't find a way to make `pre-commit` do a dry run), this has to run in a temp dir.